src/phylum/ci/ci_gitlab.py

```diff
@@ -1,39 +1,105 @@
+
+    @property
+    def gitlab_token(self) -> str:
+        """Get the GitLab token (e.g., personal, project, group, etc.)."""
+        return self._gitlab_token
+
     @property
     def phylum_label(self) -> str:
         """Get a custom label for use when submitting jobs with `phylum analyze`."""
-        return "TO DO"
+        mr_iid = os.getenv("CI_MERGE_REQUEST_IID", "unknown-IID")
+        mr_title = os.getenv("CI_MERGE_REQUEST_TITLE", "unknown-title")
+        label = f"{self.ci_platform_name}_MR#{mr_iid}_{mr_title}"
+        label = label.replace(" ", "-")
+        return label
 
-    def check_prerequisites(self) -> None:
-        """Ensure the necessary pre-requisites are met and bail when they aren't."""
-        super().check_prerequisites()
-        print("TO DO")
+    @property
+    def common_lockfile_ancestor_commit(self) -> Optional[str]:
+        """Find the common lockfile ancestor commit."""
+        # Reference: https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
+        return os.getenv("CI_MERGE_REQUEST_DIFF_BASE_SHA")
 
     def _is_lockfile_changed(self, lockfile: Path) -> bool:
         """Predicate for detecting if the given lockfile has changed."""
-        return bool("TO DO")
+        mr_src_branch = os.getenv("CI_MERGE_REQUEST_SOURCE_BRANCH_NAME")
+        mr_tgt_branch = os.getenv("CI_MERGE_REQUEST_TARGET_BRANCH_NAME")
+        mr_diff_base_sha = os.getenv("CI_MERGE_REQUEST_DIFF_BASE_SHA")
+        print(f" [+] CI_MERGE_REQUEST_SOURCE_BRANCH_NAME: {mr_src_branch}")
+        print(f" [+] CI_MERGE_REQUEST_TARGET_BRANCH_NAME: {mr_tgt_branch}")
+        print(f" [+] CI_MERGE_REQUEST_DIFF_BASE_SHA: {mr_diff_base_sha}")
+
+        # Assume no change when there isn't enough information to tell
+        if mr_diff_base_sha is None:
+            return False
 
-    def get_new_deps(self) -> Packages:
-        """Get the new dependencies added to the lockfile and return them."""
-        # TODO
+        try:
+            cmd = f"git diff --exit-code --quiet {mr_diff_base_sha} -- {lockfile.resolve()}"
+            # `--exit-code` will make git exit with with 1 if there were differences while 0 means no differences.
+            # Any other exit code is an error and a reason to re-raise.
+            subprocess.run(cmd.split(), check=True)
+            return False
+        except subprocess.CalledProcessError as err:
+            if err.returncode == 1:
+                return True
+            print(" [!] Consider changing the `GIT_DEPTH` variable in CI settings to clone/fetch more branch history")
+            raise
 
     def post_output(self) -> None:
-        """Post the output of the analysis in the means appropriate for the CI environment."""
-        # TODO: Change this placeholder when the real Gitlab CI integration is ready.
-        print(f" [+] Analysis output:\n{self.analysis_output}")
+        """Post the output of the analysis as a note (comment) on the GitLab CI Merge Request (MR)."""
```

Comment:
One limitation of this implementation is that it doesn't ensure PR comments are unique and not added multiple times as the PR changes (but the lock file doesn't). Adding that feature should be possible with the [GitLab API for Notes on MRs](https://docs.gitlab.com/ee/api/notes.html#merge-requests) ...it just got pushed back a bit due to deadlines. Is it required to ship this feature now? Or, can it be added in a different issue?

CC: @furi0us333 @peterjmorgan

Reply:
Update: I created #39 to take care of this.